cluster_tools/cluster_tools/schedulers/cluster_executor.py

```diff
@@ -38,18 +38,6 @@
 _S = TypeVar("_S")
 
 
-def _handle_kill_through_weakref(
```

Comment:
I think the same function still exists in `dask.py`. is this alright?

Reply:
@normanrz Do we still want to maintain the dask executor?

If so, it needs to be adapted as well, yes.

Reply:
Nope. The dask executor can go.

Reply:
We decided not to remove the dask executor at this point to avoid a breaking change. Not fixing this bug for it is not a big deal. We should deprecate the dask and kubernetes executors in a followup PR.